### PR TITLE
Ax 790 implement farm page analytics

### PR DESCRIPTION
**Pull request by MichaelFrank0317**
# Description
Implement Firebase Analytics on farm page.

Fixes Jira Ticket # 

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Is this a UI Change? If so please include screenshot of before and after states below

# How Has This Been Tested?
This has been tested on the UI.

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have assigned 2 reviewers to check my work
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/pages/farm/components/stake_approve_button.dart [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+// This code changes the state of the button
+class StakeApproveButton extends StatefulWidget {
+  const StakeApproveButton(
+    this.width,
+    this.height,
+    this.text,
+    this.confirmDialog,
+    this.selectedFarm, {
+    super.key,
+  });
```
Comment (Friendlygeckos) [inline lib/pages/farm/components/stake_approve_button.dart:18]: make these named parameters

File: lib/pages/farm/components/unstake_approve_button.dart [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+// This code changes the state of the button
+class UnStakeApproveButton extends StatefulWidget {
+  const UnStakeApproveButton(
+    this.width,
+    this.height,
+    this.text,
+    this.confirmDialog,
+    this.selectedFarm, {
+    super.key,
+  });
```
Comment (Friendlygeckos) [inline lib/pages/farm/components/unstake_approve_button.dart:18]: make these named parameters

File: lib/pages/farm/components/stake_approve_button.dart [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    }).catchError((_) {
+      setState(() {
+        isApproved = false;
+        text = 'Approve';
+        fillcolor = Colors.transparent;
+        textcolor = Colors.amber;
+      });
+    });
+    // Keep track of how many times the state has changed
+    currentState += 1;
```
Comment (Friendlygeckos) [inline lib/pages/farm/components/stake_approve_button.dart:98]: we do not need this

File: lib/pages/farm/components/unstake_approve_button.dart [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    final info = getUnStakeInfo();
+    context.read<TrackingCubit>().onPressedUnStake(
+      tickerPair: info.tickerPair,
+      tickerPairName: info.tickerPairName,
+      axlInput: info.axlInput,
+      axlBalance: info.axlBalance,
+    );
+    // Keep track of how many times the state has changed
+    currentState += 1;
```
Comment (Friendlygeckos) [inline lib/pages/farm/components/unstake_approve_button.dart:89]: we do not need this

**Review by Friendlygeckos — CHANGES_REQUESTED** [review]

File: lib/pages/farm/components/stake_approve_button.dart (hunk 1 above)
Comment (MichaelFrank0317) [reply lib/pages/farm/components/stake_approve_button.dart]: Made these named parameters

File: lib/pages/farm/components/unstake_approve_button.dart (hunk 2 above)
Comment (MichaelFrank0317) [reply lib/pages/farm/components/unstake_approve_button.dart]: made these named parameters

File: lib/pages/farm/components/stake_approve_button.dart (hunk 3 above)
Comment (MichaelFrank0317) [reply lib/pages/farm/components/stake_approve_button.dart]: removed this

File: lib/pages/farm/components/unstake_approve_button.dart (hunk 4 above)
Comment (MichaelFrank0317) [reply lib/pages/farm/components/unstake_approve_button.dart]: removed this

**Review by Friendlygeckos — APPROVED** [review]

**Review by pearlson16 — APPROVED** [review]

**Pull request merged**